cms/djangoapps/coursegraph/tests/test_admin.py

```diff
@@ -0,0 +1,227 @@
+"""
+Shallow tests for CourseGraph dump-queueing Django admin interface.
+
+See ..management.commands.tests.test_dump_to_neo4j for more comprehensive
+tests of dump_course_to_neo4j.
+"""
+
+from unittest import mock
+
+import py2neo
+from django.test import TestCase
+from django.test.utils import override_settings
+from freezegun import freeze_time
+
+from openedx.core.djangoapps.content.course_overviews.tests.factories import CourseOverviewFactory
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+from .. import admin, tasks
+
+
+_coursegraph_connection = {
+    "protocol": "bolt",
+    "secure": True,
+    "host": "example.edu",
+    "port": 7687,
+    "user": "neo4j",
+    "password": "fake-coursegraph-password",
+}
+
+_configure_coursegraph_connection = override_settings(
+    COURSEGRAPH_CONNECTION=_coursegraph_connection,
+)
+
+_patch_log_exception = mock.patch.object(
+    admin.log, 'exception', autospec=True
+)
+
+_patch_apply_dump_task = mock.patch.object(
+    tasks.dump_course_to_neo4j, 'apply_async'
+)
+
+_pretend_last_course_dump_was_may_2020 = mock.patch.object(
+    tasks,
+    'get_command_last_run',
+    new=(lambda _key, _graph: "2020-05-01"),
+)
+
+_patch_neo4j_graph = mock.patch.object(
+    tasks, 'Graph', autospec=True
+)
+
+_make_neo4j_graph_raise = mock.patch.object(
+    tasks, 'Graph', side_effect=py2neo.ConnectionUnavailable(
+        'we failed to connect or something!'
+    )
+)
+
+
+class CourseGraphAdminActionsTestCase(TestCase):
+    """
+    Test CourseGraph Django admin actions.
+    """
+
+    @classmethod
+    def setUpTestData(cls):
+        """
+        Make course overviews with varying modification dates.
+        """
+        super().setUpTestData()
+        cls.course_updated_in_april = CourseOverviewFactory(run='april_update')
+        cls.course_updated_in_june = CourseOverviewFactory(run='june_update')
+        cls.course_updated_in_july = CourseOverviewFactory(run='july_update')
+        cls.course_updated_in_august = CourseOverviewFactory(run='august_update')
+
+        # For each course overview, make an arbitrary update and then save()
+        # so that its `.modified` date is set.
+        with freeze_time("2020-04-01"):
```

Comment:
Comment: I don't remember seeing `freeze_time` used before. It looks really interesting, thank you!

Reply:
yay, I LOVE freeze_time!